contents/testing/integration-testing.md

```diff
@@ -0,0 +1,143 @@
+
+It is usually used for smaller applications with few components.
+
+#### [Incremental testing approach](https://www.quora.com/What-is-incremental-testing-in-software):
+
+This approach involves integrating two or more logically related components. The other related components are added and tested for proper functioning. Repeat these steps until all the components are joined and tested successfully.
+
+It is usually preferred with any bigger applications with more than 10 different components. It is further split into the 3 approaches shown below.
+
+| | [Top-down approach](https://www.guru99.com/integration-testing.html#9) | [Bottom-up approach](https://www.guru99.com/integration-testing.html#8) | [Hybrid / sandwich approach](https://www.guru99.com/integration-testing.html#10) |
+| -- | -- | -- | -- |
+| **Description**| Higher level modules are tested with lower level modules until all modules are tested. | Lower level modules are tested with higher level modules until all modules are tested. | A mix of both top-down bottom-up approaches. |
+| **Advantages** | Early discovery of high level architecture / design defects | Easier to create test cases bottom up | Beneficial for big project to distribute tasks on testing
+| | Main control points of the system are tested early | Critical modules on functionalities ae tested first | Allow top-down and bottom-up approach to run side by side | 
+| **Disadvantages** | Significant low level moduls are tested late in the cycle | There is no working system until the last module is build | It is difficult to test for highly interconnected modules |
+| | A [<span title="A program that simulates the behaviours of software components">stub</span>](https://stackoverflow.com/questions/463278/what-is-a-stub) is not perfect to simulate data flow | A [<span title="Module with dummy code to temporalities replace a module">driver</span>](http://www.professionalqa.com/test-driver) test is even harder to write than stub | Higher cost from using both driver and stub. You can understand the difference [here](https://www.quora.com/What-is-the-difference-between-stubs-and-drivers-in-software-testing)
+
+:information_source: A more detailed guide on using specific methods for incremental testing together with examples can be found [here](https://www.softwaretestinghelp.com/incremental-testing/).
+
+### Designing test scenarios, cases and scripts
+
+Before the actual coding is done, a basic test strategy deciding the test cases and test data used should be crafted. This usually involves setting a <span style="color:red">test case ID</span>, <span style="color:red">objective</span>, <span style="color:red">description</span> and <span style="color:red">expected result</span>. Using the example shown above, below shows a sample integration test used for the login and ordering modules:
```

Comment:
Use backticks for login and ordering.
Also, might be better to just insert an image snippet here since only these 2 modules are required